### PR TITLE
Undocumented/refactor zui bar diagram

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR adds some minor refactoring to the work done by @river-bbc and @lewis-conroy on creating the ZUIBarDiagram component. 

## Changes
- use the `theme` for color
- change sizes to small, medium and large
- minor styling updates

## Related issues
none

### CODE REVIEW

File: src/zui/ZUIBarDiagram/index.tsx [hunk 1]
```diff
@@ -39,11 +38,22 @@ const ZUIBarDiagram = ({ values, size }: ZUIBarDiagramProps) => {
...
+  const colors = [
+    theme.palette.data[900],
+    theme.palette.data[300],
+    theme.palette.data[100],
+  ];
+  colors.splice(
+    1,
+    0,
+    values.length == 3 ? theme.palette.data[700] : theme.palette.data[500]
+  );
```
Comment (richardolsson) [inline src/zui/ZUIBarDiagram/index.tsx:50]: I don't fully understand this logic and I'm not sure if it's just because I'm tired. Lets discuss tomorrow!

**Review by richardolsson — APPROVED** [review]
This looks good to me! There's a piece of logic here that is difficult for me to understand but that doesn't need to block the merge.

File: src/zui/ZUIBarDiagram/index.tsx (hunk 1 above)
Comment (ziggabyte) [reply src/zui/ZUIBarDiagram/index.tsx:50]: I know, it's weird af. I had a moment. 

I'll update to make it more clear, hang on.

**Review by richardolsson — APPROVED** [review]
Nice! This is much simpler for me to understand. 💯

**Pull request merged**